pkg/controller/jenkins/reconciler.go

```diff
@@ -153,14 +150,21 @@ func (r *JenkinsReconciler) Reconcile(request reconcile.Request) (reconcile.Resu
 	// TODO implement error checking for existence of the resource
 	// *or if there is an actual error* deal with it (<>_<>-)
 	// Also -- https://github.com/redhat-developer/openshift-jenkins-operator/pull/17#pullrequestreview-289463590 --
-
+	errors := new(utils.Errors)
 	err = r.createResourceIfNotPresent(jenkinsDc, jenkinsDc.Name, instance.Namespace)
+	errors.Add(err)
 	err = r.createResourceIfNotPresent(jenkinsServiceAccount, jenkinsServiceAccount.ObjectMeta.Name, instance.Namespace)
+	errors.Add(err)
```

Comment:
Should we be even moving to the next line of execution if there's an error? I would suppose we should return an error or do a `requeue on Error` right away?

Reply:
Coming back to my earlier point in https://github.com/redhat-developer/openshift-jenkins-operator/pull/35#discussion_r328866013 , this way we can have requeue in the Add method.

Reply:
@sbose78 I also agree with `requeue on Error` approach. I think we should not add a error management module, and just use more "Kubernetes native approach", which consists of requeue with any expected condition is not met.

Reply:
^ Thank you for the comment otavio. Gives me a better idea about what the architecture should look like. :thinking: